master/buildbot/db/model.py

```diff
@@ -678,6 +678,15 @@ class Model(base.DBConnectorComponent):
         sa.Column('last_active', sa.Integer, nullable=False),
     )
 
+    # gerrit change hashes
+    gerriteventhashes = sautils.Table(
```

Comment:
sourcestamp table has exactly this purpose, and is generic. Please use this table instead of adding another one.
http://docs.buildbot.net/current/developer/database.html#buildbot.db.sourcestamps.SourceStampsConnectorComponent
You will need to use findSourceStampId()

and you will need to modify it so that the returned structure properly reports that a new sourcestamp has been created (instead of having to test if created_at ~= now())

Reply:
There is also the state table, which can be used for similar purpose.
http://docs.buildbot.net/current/developer/database.html#module-buildbot.db.state

We don't add tables specific to a particular purpose.

Reply:
What is the meaning of the `codebase` parameter to the `sourcestamp`? Documentation says it's never `None` but it does not appear to be part of the gerrit `chdict` (unless it's somewhere in `properties`).

Reply:
Sorry, I don't quite understand this statement:

> and you will need to modify it so that the returned structure properly reports that a new sourcestamp has been created (instead of having to test if created_at ~= now())

The structure returned from where? Are you saying that I need to modify `findSourceStampId()` itself?

Reply:
Oh, now I see. findSourceStampId always returns an id. If it doesn't already exist, it creates one. So we need a way to know if it created one.